File: practice/python/findSingleNumber.py
class Solution:
    # @param A, a list of integer
    # @return an integer
    def _singleNumber(self,A):
        number={}
        for i in A:
            if i in number:
                number[i] += 1
            else:
                number[i] = 1
        
        for key in number:
            if number[key] == 1:
                 return key


    def singleNumber(self,A):
        sum = 0
        for i in A:
            sum ^= i
        return sum

if __name__ == '__main__':
     sol = Solution()
     print sol.singleNumber([2,2,1,1,-3,3,3,4,4,5,5,6])
     print sol.singleNumber([1])

File: practice/java/FindSingleNumber.java
public class FindSingleNumber{
    public int singleNumber(int[] A){
        int sum = 0;
        for (int i=0; i<A.length;i++){
            sum ^= A[i];
        }
        return sum;
    }
    
    public static void main(String[] argv){
        FindSingleNumber fsn = new FindSingleNumber();
        int[] A = {1,2,3,4,5,1,2,3,4,5,-4};
        int result = fsn.singleNumber(A);
        System.out.println(result);
        
    }

}



